### DISCUSSION TITLE
ESPNow, ESP32-generic sending large files to Gmail

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#11068] [ESP32]
For months I have been able to send 30-40K files using the ESPNow ESP32-SPIRAM v1.19.1 image.  However, on the ESPNow ESP32-generic v1.19.1 image I can only send 2927 bytes before I get this error:
```
Traceback (most recent call last):
  File "main.py", line 14, in <module>
  File "mail_functions.py", line 93, in send_csv
  File "my_umail.py", line 213, in write
OSError: -113
MicroPython v1.19.1-espnow-6-g44f65965b on 2022-07-09; ESP32 module with ESP32
Type "help()" for more information.
>>> 
```
in my_umail.py
```py
def write(self, content):
    print(content)
    self._sock.write(content) #  line 213
```
gc.mem_free() reports about 80K

Is there some socket size limitation on the standard Micropython ESP32-generic image c.f. the SPIRAM image?

**Comment by karfas** [comment]
A grep over the ESP-IDF sources gets me to ./components/openssl/include/internal/tls1.h:
```C
# define TLS1_AD_BAD_CERTIFICATE_STATUS_RESPONSE 113
```
So this seems to be linked to some TLS problem. But I have no idea why this should happen during `write()` - I would assume the TLS connection already exists when umail.py writes something to the socket.

**Reply by davefes** [reply]
All I could find was `OSError: [Errno 113] ECONNABORTED`
As far as I am aware it does, using port 587

Thanks for looking.
